### DISCUSSION TITLE
Why Does Oppia Android Use Older Versions?

### DISCUSSION THREAD

**Discussion by jainv4156** [oppia/oppia-android#5576] [General Discussion]
I wanted to ask why Oppia Android is using lower versions for libraries, Gradle, and other dependencies. For example, the installation guide in the [Oppia Android wiki](https://github.com/oppia/oppia-android/wiki/Installing-Oppia-Android) recommends using API level 30 for debugging, even though API level 35 is just around the corner. This makes me wonder if the wiki might not be updated to reflect the latest changes.

While I’m not complaining, I’m curious about the reasons behind this decision. One possibility I could think of is ensuring compatibility with older devices. Another might be that upgrading all dependencies requires significant effort. I could be wrong, but I’d like to understand the rationale better. Could you clarify?

**Comment by adhiamboperes** [comment]
@jainv4156, great question!

We use relatively older dependencies because it would require significant effort to update the current ones, and test the app thoroughly for breaking changes, which are likely.

Oppia Android runs on a relatively older Android Gradle Plugin, and for a while, we also used an older version of Bazel. This sometimes led to compatibility issues when updating dependency versions. However, our current Bazel version is now pretty new, and we will be able to slowly update the dependencies once the Gradle - Bazel migration is complete.
